semgrep/semgrep/core_runner.py

```diff
@@ -199,99 +199,98 @@ def _run_rule(
         outputs: List[PatternMatch] = []  # multiple invocations per language
         errors: List[Any] = []
         equivalences = rule.equivalences
-        with tempfile.NamedTemporaryFile("w") as equiv_fout:
 
-            if equivalences:
-                self._write_equivalences_file(equiv_fout, equivalences)
+        for language, all_patterns_for_language in self._group_patterns_by_language(
```

Comment:
Most of this diff is deindenting

Reply:
Hah nice, these loops were getting a bit unwieldy.